log/logger_multiwriter.go

```diff
@@ -22,36 +28,66 @@ func loggerWorker() {
 	buffer := make([]byte, 0, defaultBufferCapacity)
 	var n int
 	var err error
+
+	structuredOutbound := ExtraFields{}
 	for j := range jobsChannel {
 		if j.Passback != nil {
 			j.Passback <- struct{}{}
 			continue
 		}
-		data := j.fn()
-		buffer = append(buffer, j.Header...)
-		if j.ShowLogSystemName {
-			buffer = append(buffer, j.Spacer...)
-			buffer = append(buffer, j.SlName...)
-		}
-		buffer = append(buffer, j.Spacer...)
-		if j.TimestampFormat != "" {
-			buffer = time.Now().AppendFormat(buffer, j.TimestampFormat)
-		}
-		buffer = append(buffer, j.Spacer...)
-		buffer = append(buffer, data...)
-		if data == "" || data[len(data)-1] != '\n' {
+		msg := j.fn()
+		if j.StructuredLogging {
+			structuredOutbound[message] = msg
+			structuredOutbound[timestamp] = time.Now().UnixMilli()
+			structuredOutbound[severity] = j.Severity
+			structuredOutbound[subLoggerName] = j.SubLoggerName
+			structuredOutbound[botName] = j.Instance
+			for k, v := range j.StructuredFields {
+				_, ok := structuredOutbound[k]
+				if ok {
+					// Disallow overwriting of key values
+					displayError(fmt.Errorf("structured logging: cannot overwrite key [%s]", k))
+					continue
+				}
+				structuredOutbound[k] = v
+			}
+			buffer, err = json.Marshal(structuredOutbound)
+			if err != nil {
+				log.Println("log: failed to marshal structured log data:", err)
+			}
+			for k := range j.StructuredFields {
+				// Delete non-persistent structured fields
+				delete(structuredOutbound, k)
+			}
```

Comment:
Nitpick query: Would it not be easier to just do:
```
structuredOutbound = ExtraFields{}
```

Reply:
Code wise it is much nicer but I want to preserve the allocation and default keys within that map. Squeeze out some nanos. 
<img width="440" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9391715/a1a089b8-07f9-4df2-bce8-a2d22079a059">